### PR TITLE
Better theme palette generation

### PR DESCRIPTION
**Pull request by AdityaHegde**
This is a prototype with an updated algorithm to get the color palette from the input color.

- Find the color with the lowest [distance](https://www.vis4.net/chromajs/#chroma-distance) with any color in the list of [tailwind colors](https://tailwindcss.com/docs/customizing-colors)
- Calculate the shift in hue and saturation of the matched color compared to input color.
- Get the palette this matched color belongs to and apply the shift to all colors.

Primary color is now available in the dashboard. Secondary color is available as well since we updated the spinner to be primary-500 to secondary-500.

### CODE REVIEW

**Comment by jkhwu** [comment]
@AdityaHegde I cloned this branch, ran the app locally, but http://localhost:3000/dev/color-palette didn't go anywhere. I'll Slack you for help.

**Comment by jkhwu** [comment]
Checked the theme color palette and looks good, will check back in after it's applied to dashboards. Thanks @AdityaHegde !

**Comment by bcolloran** [comment]
@AdityaHegde and @jkhwu -- request that we merge this after https://github.com/rilldata/rill/pull/3863

**Comment by jkhwu** [comment]
Just checked these colors applied to the dashboard charts and pills (without the spinner updates). They look good to me.

File: web-common/package.json
Comment (bcolloran) [inline web-common/package.json:45]: can we use either colorjs.io or chroma-js? I don't think we need both.

File: web-local/src/routes/dev/color-palette/+page.svelte
Comment (bcolloran) [inline web-local/src/routes/dev/color-palette/+page.svelte:22]: Can we remove this try/catch? None of these functions seem like they should be fallible. If one is, then we should try to debug it. If the try/catch is absolutely needed because a library function from chroma.js is failing let's remove that dep and use colorio.js

File: web-common/src/features/themes/color-config.ts
Comment (bcolloran) [inline web-common/src/features/themes/color-config.ts:317]: no big deal, but this could flatten down to single defintion:
```suggestion
export const TailwindColorPresetsConverted: Color[][] = [];
for (const colorName in TailwindColorPresets) {
  const colors: Color[] = [];
  for (const shade of TailwindColorPresets[colorName]) {
    colors.push(chroma(shade));
  }
  TailwindColorPresetsConverted.push(colors);
}
```

```suggestion
export const TailwindColorPresetsConverted: Color[][] = TailwindColorPresets.map(
  (name) =>  TailwindColorPresets[name].map((shade) =>chroma(shade))
)
```

**Review by bcolloran — APPROVED** [review]
Looks good, a couple small change request that you can handle as you see fit @AdityaHegde, but I'm going to mark this as approved, I don't feel the need for a second review (but am happy to review again if you  want to discuss anything)

File: web-common/package.json
Comment (AdityaHegde) [reply web-common/package.json:45]: chroma-js has a method that was not in colorjs.io. I can remove colorjs.io

**Pull request merged**
